### DISCUSSION TITLE
Is there support for network devices with context/vDom etc? Like Cisco ASA, Fortigate, PaloAlso.

### DISCUSSION THREAD

**Discussion by empusas** [scrapli/scrapli_community#181] [Q&A]
I did look at the code and could not find any support to switch the configuration context on devices that do support context or vDom or however it is called on the platform.
I could write my own wrapper to switch and keep state of the current context. But I would rather add that support into the module itself.

Any comments if such a PR would be welcome?

**Comment by carlmontanari** [comment]
hey @empusas 👋 

you *should* be able to handle this just with custom privilege levels I think -- you can find the main class [here](https://github.com/carlmontanari/scrapli/blob/main/scrapli/driver/network/base_driver.py#L21) -- and you can find these implemented all over scrapli community. you can set "escalate" to some command like "go to my context" (or whatever obviously im just making stuff up :p) and ensure you have a prompt pattern that matches the context prompt. 

if there is "extra" stuff that needs to happen that cant be done with priv levels I *might* merge a pr for that depending on how elaborate it is -- generally scrapli community things try to stay pretty simple and just act as a collection of priv levels/prompts that work for various devices.
